packages/health-check/src/checks/critical.js

```diff
@@ -0,0 +1,48 @@
+const superagent = require("superagent");
+const { StatusCodes } = require("http-status-codes");
+const { getTimeDiff } = require("../utils");
+
+// uploadCheck returns the result of uploading a sample file
+async function uploadCheck(done) {
+  const time = process.hrtime();
+
+  superagent
+    .post(`http://${process.env.PORTAL_URL}/skynet/skyfile`)
+    .attach("file", "package.json", "package.json")
+    .end((error, response) => {
+      const statusCode = (response && response.statusCode) || (error && error.statusCode) || null;
+
+      done({
+        name: "upload_file",
+        up: statusCode === StatusCodes.OK,
+        statusCode,
+        time: getTimeDiff(time),
+        critical: true,
+      });
+    });
+}
+
+// downloadCheck returns the result of downloading the hard coded link
+async function downloadCheck(done) {
+  const time = process.hrtime();
+  const skylink = "AACogzrAimYPG42tDOKhS3lXZD8YvlF8Q8R17afe95iV2Q";
+  let statusCode;
```

Comment:
(nit) more tidy
```
let up;

  try {
        const response = await superagent.get(`http://${process.env.PORTAL_URL}/${skylink}?nocache=true`);
        up = response.statusCode === StatusCodes.OK;
  } catch (error) {
        up = false'
  }

  done({
    name: "download_file",
    up,
    time: getTimeDiff(time),
    critical: true,
  });
```

Reply:
we still need `statusCode` in the response

Reply:
it would also be nice to rewrite the download to match the structure of the upload, why isn't that possible? would be much cleaner

Reply:
> we still need statusCode in the response

I missed that!

Reply:
> it would also be nice to rewrite the download to match the structure of the upload, why isn't that possible? would be much cleaner

because upload is a stream and download is a promise